verification/README.md

```diff
@@ -0,0 +1,210 @@
+```
+In that case, you can run the binary using just `verification`.
+
+## Configuration
+Currently, the service uses a configuration file the path to which is specified via CLI flag `--config-path=[path]`.
+The configuration file may contain the following options:
+```toml
+[server]
+# IP address and port number the server should listen to
+addr = "0.0.0.0:8043"
+
+[solidity]
+# if false solidity verification is not supported
+enabled = true
+# list of all available compilers and information about them
+compilers_list_url = "https://raw.githubusercontent.com/blockscout/solc-bin/main/list.json"
+
+[sourcify]
+# if false sourcify verification is not supported
+enabled = true
+# Sourcify API endpoint
+api_url = "https://sourcify.dev/server/"
+# number of failing attempts the server makes to Sourcify API 
+verification_attempts = 3
+# the maximum period the service is waiting for the Sourcify response
+request_timeout = 10
+```
+For all keys omitted from the configuration file default values from the example above are used.
+
+# Api
+## Input
+Currently, the service supports 4 types of verification:
+
+### Single file
+**Note**: Is deprecated and going to be replaced by Multi-Part files verification
+
+```json
+{
+  // Creation transaction input
+  "creation_bytecode": "0x608060...0033000b0c",
+  // Bytecode stored in the blockchain
+  "deployed_bytecode": "0x608060...0033",
+  // Compiler version used to compile the contract
+  "compiler_version": "v0.8.14+commit.80d49f37",
+  // Source code
+  "source_code": "pragma solidity ^0.8.14; ...",
+  // Version of the EVM to compile for
+  "evm_version": "default",
+  // If present, optimizations are enabled with specified number of runs, 
+  // otherwise optmimizations are disabled
+  "optimization_runs": "200",
+  // If present, specify addresses of the libraries.
+  "contract_libraries": {
+    "MyLib": "0x123123..."
+  }
+}
+```
+
+### Multi-Part files
+**Note**: currently WIP and is not available right now
+
+The only difference with Single file input is that the `source_code` field was replaced by `sources` allowing to submit several files for verification.
+```json
+{
+  // Creation transaction input
+  "creation_bytecode": "0x608060...0033000b0c",
+  // Bytecode stored in the blockchain
+  "deployed_bytecode": "0x608060...0033",
+  // Compiler version used to compile the contract
+  "compiler_version": "v0.8.14+commit.80d49f37",
+  // Contains a map from a source file name to the actual source code
+  "sources": {
+    "A.sol": "pragma solidity ^0.8.14; contract A {}",
+    "B.sol": "pragma solidity ^0.8.14; contract B {}"
+  },
+  // Version of the EVM to compile for
+  "evm_version": "default",
+  // If present, optimizations are enabled with specified number of runs, 
+  // otherwise optmimizations are disabled
+  "optimization_runs": "200",
+  // If present, specify addresses of the libraries.
+  "contract_libraries": {
+    "MyLib": "0x123123..."
```

Comment:
I think we should accept double map, like

```json
"contract_libraries": {
  "A.sol": {
    "MyLib": "0x123123..."
  }
}
```

Reply:
We've decided to accept single level map here, as usually users do not specify what file the library belongs to. We will handle different cases internally, by bruteforcing different combinations of file and library names